File: 3779.minimum-number-of-operations-to-have-distinct-elements.cpp
#
# @lc app=leetcode id=3779 lang=cpp
#
# [3779] Minimum Number of Operations to Have Distinct Elements
#
# @lc code=start
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int operations = 0;
        int start = 0;
        int n = nums.size();
        
        while (start < n) {
            // Check if remaining elements are all distinct
            unordered_set<int> seen;
            bool hasDuplicate = false;
            for (int i = start; i < n; i++) {
                if (seen.count(nums[i])) {
                    hasDuplicate = true;
                    break;
                }
                seen.insert(nums[i]);
            }
            
            if (!hasDuplicate) {
                break;
            }
            
            // Remove first 3 elements (or all remaining if < 3)
            start += 3;
            operations++;
        }
        
        return operations;
    }
};
# @lc code=end

File: 3779.minimum-number-of-operations-to-have-distinct-elements.java
#
# @lc app=leetcode id=3779 lang=java
#
# [3779] Minimum Number of Operations to Have Distinct Elements
#
# @lc code=start
class Solution {
    public int minOperations(int[] nums) {
        int n = nums.length;
        int ops = 0;
        int start = 0;
        while (start < n) {
            // Check if the current subarray is all distinct
            java.util.Set<Integer> set = new java.util.HashSet<>();
            for (int i = start; i < n; i++) set.add(nums[i]);
            if (set.size() == n - start) break;
            // Remove up to first three elements
            start = Math.min(n, start + 3);
            ops++;
        }
        return ops;
    }
}
# @lc code=end